keys/signer_ed25519.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Coinbase, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package keys
+
+import (
+	"crypto/ed25519"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/coinbase/rosetta-sdk-go/asserter"
+	"github.com/coinbase/rosetta-sdk-go/types"
+)
+
+// SignerEd25519 is initialized from a keypair
+type SignerEd25519 struct {
+	KeyPair KeyPair
+}
+
+// PublicKey returns the PublicKey of the signer
+func (s SignerEd25519) PublicKey() types.PublicKey {
+	return *s.KeyPair.PublicKey
+}
+
+// Signs arbitrary payloads using a KeyPair
+func (s SignerEd25519) Sign(payload types.SigningPayload) (*types.Signature, error) {
+	err := s.KeyPair.IsValid()
+	if err != nil {
+		return nil, err
+	}
+	privKeyBytes := s.KeyPair.PrivateKey.Bytes
+	decodedMessage, err := hex.DecodeString(payload.HexBytes)
+	if err != nil {
+		return nil, fmt.Errorf("sign: unable to decode message. %w", err)
+	}
+
+	privKey := ed25519.NewKeyFromSeed(privKeyBytes)
+	sig := ed25519.Sign(privKey, decodedMessage)
```

Comment:
I'd add an assertion here that the `SigningPayload.SignatureType` is `empty` or that it is `ed25519` : https://github.com/coinbase/rosetta-sdk-go/pull/54/files#r449318027